src/clm/functions.py

```diff
@@ -451,16 +451,67 @@ def get_mass_range(mass, err_ppm):
     return min_mass, max_mass
 
 
-def write_to_csv_file(file_name, df):
-    dirname = os.path.dirname(file_name)
-    if dirname:
-        os.makedirs(dirname, exist_ok=True)
-    df.to_csv(
+def write_to_csv_file(
+    file_name,
+    info,
+    mode="w",
+    header=True,
+    columns=None,
+    string_format=None,
+):
+    # Make an output directory if it doesn't yet
+    os.makedirs(os.path.dirname(file_name), exist_ok=True)
```

Comment:
need to take care of `if dirname` if there's no directory component to file_name